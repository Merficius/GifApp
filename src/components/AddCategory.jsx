import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const changeInput = ({ target }) => {
    setInputValue(target.value);
  };

  const submit = (event) => {
    event.preventDefault();

    const trimmedInputValue = inputValue.trim();
    if (trimmedInputValue.length < 1) return;

    onNewCategory(trimmedInputValue);

    setInputValue('');
  };

  return (
    <form action="" onSubmit={submit} aria-label="form">
      <input type="text" placeholder="Search Gifs" value={inputValue} onChange={changeInput} />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
