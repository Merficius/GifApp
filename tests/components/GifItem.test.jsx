import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

const title = 'Marvel';
const url = 'https://google.com/';

describe('GifItem tests', () => {
  test('Component snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test('Image rendered with URL and ALT attributtes', () => {
    render(<GifItem title={title} url={url} />);

    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('Title being shown in component', () => {
    render(<GifItem title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy();
  });
});
