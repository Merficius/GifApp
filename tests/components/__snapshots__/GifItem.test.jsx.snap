// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem tests Component snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Marvel"
      src="https://google.com/"
    />
    <p>
      Marvel
    </p>
  </div>
</div>
`;
